include/solvers/mpm_base.tcc

```diff
@@ -329,6 +329,26 @@ bool mpm::MPMBase<Tdim>::initialise_particles() {
                        particles_volume_end - particles_volume_begin)
                        .count());
 
+    // Material id update using particle sets
+    try {
+      auto material_sets = io_->json_object("material_sets");
```

Comment:
We need to check if the `material_sets` is available. This should be an optional argument

Reply:
How do we check availability for `material_sets`?

`mpm` successfully runs even if `material_sets` is not provided in input `JSON`.  So I think this already implemented as an optional argument? 

![image](https://user-images.githubusercontent.com/62029065/88222410-cdab6800-cc1a-11ea-97fb-3855655be518.png)

Reply:
Oh you are using `io_->json_object()` function call, which does that check for you, so you are right, no need to do it again. 

```
  if (json_.find(key) != json_.end()) {
    return json_.at(key);
  } else {
    throw std::runtime_error("No object found, returning an empty object");
    return empty;
  }
```